### PR TITLE
Random123 as a submodule

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: extra/nrnivmodl_core_makefile.in [hunk 1]
```diff
@@ -30,6 +30,7 @@ _cm =,
 
 # We rebuild the include dirs since a lot of stuff changes place
 INCLUDES = $(INCFLAGS) -I$(incdir) -I$(incdir)/coreneuron/utils/randoms
+INCLUDES += -I@CORENEURON_PROJECT_SOURCE_DIR@/external/Random123/include
```
Comment (ohm314) [inline extra/nrnivmodl_core_makefile.in]: will this continue working after `make install` ? I think we need to make sure that random123 is where we expect it to be and the generated makefile shoudl be able to find it when we run `nrnivmodl-core` from the installation target.

Comment (pramodk) [reply extra/nrnivmodl_core_makefile.in]: I don't think we have any mod file in tests that use Random123. Would be good idea to add that. I can help.

**Comment by pramodk** [comment]
I believe I/we have made (very small) changes to PGI, Fujitsu and Cray compiler headers. We need review those before merging those.

@alkino : if you look at git log of `features` directory, it should be there?

**Comment by pramodk** [comment]
Please retest

File: extra/nrnivmodl_core_makefile.in (hunk 1 above)
Comment (pramodk) [reply extra/nrnivmodl_core_makefile.in]: > will this continue working after make install ? 
As soon as you remove source, it won't work. I fixed this.

**Comment by pramodk** [comment]
Please retest

**Pull request merged**